spec/features/badge_spec.rb

```diff
@@ -61,20 +49,58 @@ def log_user(login, password)
       check 'pref_issue_badge'
       click_on 'Save'
 
-      expect(page).to have_selector("#issue_badge")
+      expect(page).to have_selector('#issue_badge')
+    end
 
-      user = User.where(:login => 'dlopper').first
-      all_issues = Issue.visible.open.where(:assigned_to_id => ([user.id] + user.group_ids))
-      puts "dlopper's issue: #{all_issues.length}"
-      if all_issues.any?
-        expect(page).to have_selector("#issue_badge_number")
-        expect(page).to have_selector("#issue_badge_number"), text: all_issues.length
+    scenario 'Badge number is displayed if badge option is activated and operator has assigned issues.' do
+      # Enable Badge
+      check 'pref_issue_badge'
+      click_on 'Save'
 
-        puts "Badge count: " + find(:id, 'issue_badge_number').text
-        page.save_screenshot('capture/badge.png', :full => true)
+      user = User.where(login: 'dlopper').first
+      all_issues = Issue.visible.open.where(assigned_to_id: ([user.id] + user.group_ids))
+      if all_issues.any?
+        expect(page).to have_selector('#issue_badge_number'), text: all_issues.length
+        page.save_screenshot('capture/badge.png', full: true)
       end
     end
+  end
 
+  context 'When Administrator' do
+    background do
+      log_user('admin', 'admin')
+      visit '/settings/plugin/redmine_issue_badge'
+    end
 
+    scenario 'Badge is not displayed if global settings badge option is not activated.' do
+      expect(page).not_to have_selector('#issue_badge')
+    end
+
+    scenario 'Badge is displayed if global settings badge option is activated.' do
+      assert page.has_content?('Display issue badge for all users')
+
+      # Enable Badge
+      check 'settings_activate_for_all_users'
+      click_on 'Apply'
+      expect(page).to have_selector('#issue_badge')
+    end
+
+    scenario 'Issue badge block is displayed if global settings badge option is activated and click badge.' do
```

Comment:
__[rubocop]__
* Line is too long. [110/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)